src/test/rustdoc-ui/doc-alias-assoc-const.rs

```diff
@@ -0,0 +1,48 @@
+#![feature(doc_alias)]
+#![feature(trait_alias)]
+
+pub struct Foo;
+
+pub trait Bar {
+    const BAZ: u8;
+}
+
+impl Bar {
+    #[doc(alias = "CONST_BAZ2")] //~ ERROR
```

Comment:
Associated consts on inherent impls do appear in the search index even if they're for dyn traits so this shouldn't be an error.

Reply:
Not on trait implementations apparently. Give it a try locally, you can find `BAZ` in the search index.

Reply:
I see it just fine, as long as I add `pub`. The `BAZ` in this impl is not related in any way to the `BAZ` on the trait. In fact this code shouldn't even be accepted in rustdoc because `Bar` isn't object safe. rustc rejects this impl.

Reply:
So if they are rejected, it's fine I guess?